### PR TITLE
updated pom and server.xml for 2.0 plugin

### PR DESCRIPTION
**Pull request by mattbsox**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -70,44 +67,65 @@
...
+            </plugin>
             <plugin>
                 <groupId>net.wasdev.wlp.maven.plugins</groupId>
                 <artifactId>liberty-maven-plugin</artifactId>
+                <version>2.0-SNAPSHOT</version>
                 <configuration>
                     <configFile>src/main/wlp/server.xml</configFile>
+                    <installAppPackages>project</installAppPackages>
+                    <looseApplication>true</looseApplication>
+                    <type>zip</type>
```
Comment (patricktiu) [inline pom.xml]: You can delete line 90 `<type>zip</zip>` which is only used within the `<install>` or `<assemblyArtifact>` element.

see https://github.com/WASdev/ci.maven/blob/master/docs/installation-configuration.md

**Review by patricktiu — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (mattbsox) [reply pom.xml]: Okay, I'll fix that in the rest of the samples too.

File: pom.xml [hunk 2]
```diff
@@ -34,80 +34,92 @@
...
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <scope>provided</scope>
+            <version>3.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.transaction</groupId>
+            <artifactId>javax.transaction-api</artifactId>
+            <version>1.2-b03</version>
```
Comment (patricktiu) [inline pom.xml:53]: This is a problem in the original sample.  javax.transaction-api dependency should have `provided` scope.

File: pom.xml [hunk 3]
```diff
@@ -124,19 +136,14 @@
                                     <groupId>org.apache.derby</groupId>
                                     <artifactId>derby</artifactId>
                                     <overWrite>false</overWrite>
-                                    <outputDirectory>${project.build.directory}/shared/resources/derby</outputDirectory>
+                                    <outputDirectory>${project.build.directory}/liberty/wlp/usr/shared/resource/derby</outputDirectory>
```
Comment (patricktiu) [inline pom.xml]: typo:

${project.build.directory}/liberty/wlp/usr/shared/`resources`/derby

File: src/main/wlp/server.xml [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+        <feature>jdbc-4.1</feature>
+        <feature>jndi-1.0</feature>
+        <feature>cdi-1.2</feature>
+    </featureManager>
 
+	<httpEndpoint host="*" httpPort="9080" httpsPort="9443" id="defaultHttpEndpoint"/>
 
-<httpEndpoint host="*" httpPort="9080" httpsPort="9443" id="defaultHttpEndpoint"/>
+	<library id="DerbyLib">
+        <file name="${wlp.user.dir}/shared/resource/derby/derby.jar"/>
```
Comment (patricktiu) [inline src/main/wlp/server.xml]: change it back to 

`<file name="${shared.resource.dir}/derby/derby.jar"/>`

See KC for the value of ${shared.resource.dir}:
https://www.ibm.com/support/knowledgecenter/SS7K4U_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/rwlp_dirs.html

File: src/main/wlp/server.xml [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
-<dataSource id="DefaultDataSource">
-<jdbcDriver libraryRef="DerbyLib"/>
-<properties.derby.embedded createDatabase="create" databaseName="${shared.resource.dir}/derby/JTASAMPLE"/>
-</dataSource>
-<library id="DerbyLib">
-<file name="${shared.resource.dir}/derby/derby.jar"/>
-</library>
+    <dataSource id="DefaultDataSource" jndiName="jtaDB">
+        <jdbcDriver libraryRef="DerbyLib"/>
+        <properties.derby.embedded createDatabase="create" databaseName="${wlp.user.dir}/shared/resource/derby/JTASAMPLE"/>
```
Comment (patricktiu) [inline src/main/wlp/server.xml]: change it back to 

`<properties.derby.embedded createDatabase="create" databaseName="${shared.resource.dir}/derby/JTASAMPLE"/>`

**Review by patricktiu — CHANGES_REQUESTED** [review]

**Review by patricktiu — APPROVED** [review]

**Review by patricktiu — APPROVED** [review]

**Pull request merged**